pkg/framework/util.go

```diff
@@ -206,9 +212,205 @@ func ListPodSanboxForID(c internalapi.RuntimeService, podID string) []*runtimeap
+}
+
+// StopContainer stops the container for containerID.
+func StopContainer(c internalapi.RuntimeService, containerID string, timeout int64) {
+	By("Stop container for containerID: " + containerID)
+	stopped := make(chan bool, 1)
+
+	go func() {
+		err := c.StopContainer(containerID, timeout)
+		ExpectNoError(err, "failed to stop container: %v", err)
+		stopped <- true
+	}()
+
+	select {
+	case <-time.After(time.Duration(timeout) * time.Second):
+		Failf("stop container %q timeout.\n", containerID)
+	case <-stopped:
+		Logf("Stopped container %q\n", containerID)
+	}
+}
+
+// TestStopContainer stops the container for containerID and make sure it's exited.
+func TestStopContainer(c internalapi.RuntimeService, containerID string) {
+	StopContainer(c, containerID, defaultStopContainerTimeout)
+	VerifyContainerStatus(c, containerID, runtimeapi.ContainerState_CONTAINER_EXITED, "exited")
+}
+
+// RemoveContainer removes the container for containerID.
+func RemoveContainer(c internalapi.RuntimeService, containerID string) {
+	By("Remove container for containerID: " + containerID)
+	err := c.RemoveContainer(containerID)
+	ExpectNoError(err, "failed to remove container: %v", err)
+	Logf("Removed container %q\n", containerID)
+}
+
+// ListContainerForID lists container for containerID.
+func ListContainerForID(c internalapi.RuntimeService, containerID string) []*runtimeapi.Container {
+	By("List containers for containerID: " + containerID)
+	filter := &runtimeapi.ContainerFilter{
+		Id: containerID,
+	}
+	containers, err := c.ListContainers(filter)
+	ExpectNoError(err, "failed to list containers %q status: %v", containerID, err)
+	return containers
+}
+
+// CreatePodSandboxForContainer creates a PodSandbox for creating containers.
+func CreatePodSandboxForContainer(c internalapi.RuntimeService) (string, *runtimeapi.PodSandboxConfig) {
+	podSandboxName := "create-PodSandbox-for-container-" + NewUUID()
+	uid := defaultUIDPrefix + NewUUID()
+	namespace := defaultNamespacePrefix + NewUUID()
+	config := &runtimeapi.PodSandboxConfig{
+		Metadata: buildPodSandboxMetadata(podSandboxName, uid, namespace, defaultAttempt),
+	}
+
+	podID := RunPodSandbox(c, config)
+	return podID, config
+}
+
+// TestExecSync test execSync for containerID and make sure the response is right.
+func TestExecSync(c internalapi.RuntimeService, containerID string) {
+	By("Test execSync for containerID: " + containerID)
+	cmd := []string{"echo", "hello"}
+	stdout, stderr, err := c.ExecSync(containerID, cmd, time.Duration(defaultExecSyncTimeout)*time.Second)
+	ExpectNoError(err, "failed to execSync in container %q", containerID)
+	Expect(string(stdout)).To(Equal("hello\n"), "The stdout output of execSync should be %q", stdout)
+	Expect(stderr).To(BeNil(), "The stderr should be nil.")
+	Logf("Execsync succeed")
+}
+
+// PullPublicImage pulls the public image named imageName.
+func PullPublicImage(c internalapi.ImageManagerService, imageName string) {
```

Comment:
Some functions in `framework.go` will need to call these image related functions, such as `CreateContainer`.